examples/object-oriented-cpp/PineAPPL.hpp

```diff
@@ -22,6 +21,14 @@ struct KeyVal {
     /** @brief constructor */
     KeyVal() : raw(pineappl_keyval_new()) {}
 
+    KeyVal(const KeyVal&) = delete;
+
+    KeyVal(KeyVal&&) = delete;
+
+    KeyVal& operator=(const KeyVal&) = delete;
+
+    KeyVal& operator=(KeyVal&&) = delete;
+
```

Comment:
Since we have a pointer as a member all other constructors and assignment operators should be deleted; otherwise you can copy the C++ object that doesn't really copy `raw`.